misc/src/main/java/com/redhat/lightblue/savedsearch/FindRequestBuilder.java

```diff
@@ -0,0 +1,256 @@
+
+    /**
+     * Given the parameter values supplied by the caller, returns a
+     * map containing default values for missing parameters. If there
+     * are missing required parameters, throws an Error with missing
+     * parameters.
+     */
+    public static Map<String,String> fillDefaults(JsonNode savedSearch,
+                                                  Map<String,String> parameterValues,
+                                                  TypeResolver types) {
+        Map<String,String> ret=new HashMap<>();
+        JsonNode parametersNode=savedSearch.get("parameters");
+        if(parametersNode instanceof ArrayNode) {
+            for(Iterator<JsonNode> itr=parametersNode.elements();itr.hasNext();) {
+                JsonNode parameterNode=itr.next();
+                if(parameterNode instanceof ObjectNode) {
+                    Type type=null;
+                    JsonNode x=parameterNode.get("name");
+                    String name=x.asText();
+                    x=parameterNode.get("type");
+                    if(x!=null) {
+                        type=types.getType(x.asText());
+                        if(type==null) {
+                            throw Error.get(ERR_SAVED_SEARCH_INVALID_TYPE,x.asText());
+                        }
+                    } else {
+                        type=StringType.TYPE;
+                    }
+                    boolean optional=false;
+                    x=parameterNode.get("optional");                    
+                    if(x!=null) {
+                        optional=x.asBoolean();
+                    }
+                    if(parameterValues.containsKey(name)) {
+                        ret.put(name,(String)StringType.TYPE.cast(type.cast(parameterValues.get(name))));
+                    } else if(optional) {
+                        // Fill it in using default value                    
+                        JsonNode defaultValue=parameterNode.get("defaultValue");
+                        if(defaultValue==null)
+                            throw Error.get(ERR_SAVED_SEARCH_NO_DEFAULT_VALUE,name);
+                        ret.put(name,(String)StringType.TYPE.cast(type.fromJson(defaultValue)));
+                    } else {
+                        throw Error.get(ERR_SAVED_SEARCH_MISSING_PARAM,name);
+                    }                    
+                }
+            }
+        }
+        return ret;
+    }
+
+    /**
+     * Builds a find request from the saved search by rewriting the
+     * search bound parameters using parameter values. The search
+     * query components can be either strings, or JSON objects. If
+     * they are strings, the parameters are replaced first, then the
+     * resulting json string is converted to a query/projection. If
+     * the components are JSON objects, then the values of those json
+     * objects are replaced with parameter values.
+     */
+    public static FindRequest buildRequest(JsonNode savedSearch,
```

Comment:
The implementation is in rest, I'll push the PR once this is in. The flow is: parse URL, get saved search from the cache, build request using url params, and run the search.